src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -249,13 +258,18 @@ private static RuntimeConfig HandleCosmosNoSqlConfiguration(string? schema, Runt
             options[namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Database))] = JsonSerializer.SerializeToElement(database);
         }
 
+        dataSource = dataSource with { Options = options, ConnectionString = connectionString };
+
         // Update the connection string in the parsed config with the one that was provided to the controller
```

Comment:
This comment should be moved up, since the connectionstring replacement happens on line 261 now.

Reply:
fixed.